### PR TITLE
Add parent ids to our reporting group table

### PR DESCRIPTION
**Pull request by jon-betts**
For:

 * https://github.com/orgs/hypothesis/projects/102

This modifies the `report.groups` to include a parent id. This will allow us to try and roll up counts as a next step based on this.

### CODE REVIEW

File: lms/data_tasks/report/create_from_scratch/02_entities/03_groups/03_groups/01_create_view.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+-- `report.raw_group` it is based on by adding `parent_id` information.
+
+CREATE MATERIALIZED VIEW report.groups AS (
+    SELECT
+        raw_groups.id,
+        parent_ids.parent_id,
+        raw_groups.authority_provided_id,
+        raw_groups.name,
+        raw_groups.group_type,
+        raw_groups.created
```
Comment (jon-betts) [inline lms/data_tasks/report/create_from_scratch/02_entities/03_groups/03_groups/01_create_view.sql:14]: We could do `raw_groups.*`, but I prefer spelling out the columns and it allows us to put the id and parent_id next to each other.

File: lms/data_tasks/report/create_from_scratch/02_entities/03_groups/01_raw_groups/01_create_view.jinja2.sql [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+-- ==============================
+-- The authority provided id is based on a number of things:
+--
+--   * The GUID
+--   * The unique id provided by the LMS for the grouping
+--   * Our type (course, section etc.)
+--
+-- It is *not* based on the application instance id at all.
+
+CREATE VIEW report.raw_groups AS (
```
Comment (jon-betts) [inline lms/data_tasks/report/create_from_scratch/02_entities/03_groups/01_raw_groups/01_create_view.jinja2.sql:17]: Note this is not a `MATERIALIZED` view here. I'm hoping this is performant enough for now.

File: lms/data_tasks/report/create_from_scratch/02_entities/03_groups/03_groups/01_create_view.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    -- Here we do a join across a lot of things to generate the parent group id
+    -- if there is one, from the grouping table parent id
+    LEFT OUTER JOIN (
+        -- We might have multiple copies of the heirarchy in the grouping table
+        -- from the perspective of different application instances. They should
+        -- all line up, but just in case they don't we group by group id and
+        -- pick the minimum `parent_id`
+        SELECT
+            raw_groups.id AS group_id,
+            MIN(parent_group_map.group_id) AS parent_id
```
Comment (marcospri) [inline lms/data_tasks/report/create_from_scratch/02_entities/03_groups/03_groups/01_create_view.sql:26]: If they are different `max` might be newer but not sure what that means in practice, probably not more correct.

Comment (jon-betts) [reply lms/data_tasks/report/create_from_scratch/02_entities/03_groups/03_groups/01_create_view.sql:26]: Yeah, I don't know. Which every you pick seems wrong?

**Pull request merged**